components/eamxx/src/control/atmosphere_driver.cpp

```diff
@@ -685,6 +699,23 @@ void AtmosphereDriver::create_fields()
     fm->add_to_group(fid.name(),"RESTART");
   }
 
+  auto& driver_options_pl = m_atm_params.sublist("driver_options");
+  const int verb_lvl = driver_options_pl.get<int>("atmosphere_dag_verbosity_level",-1);
+  if (verb_lvl>0) {
+    // now that we've got fields, generate a DAG with fields and dependencies
+    // NOTE: at this point, fields provided by initial conditions may (will)
+    // appear as unmet dependencies
+    AtmProcDAG dag;
+
+    // First, add all atm processes
+    dag.create_dag(*m_atm_process_group);
```

Comment:
I'm not too familiar with the DAG, but would it make sense to have a class variable for the driver
```
std::shared_ptr<AtmProcDAG> m_dag;
```
and then whenever we have the `m_atm_process_group` setup, call something like 
```
if (verb_lvl>0) {
  auto m_dag = std::make_shared<AtmProcDAG>();
  m_dag->create_dag(*m_atm_process_group);
}
```
Then in all these places we only need to call `m_dag->write_dag(...)`.

Reply:
I like this idea in principal, however I do think it would create a small headache spot with regard to how initialization happens with the PG2 grid (this is partly discussed in #3018). I haven't chased it down to the exact workings, but at some point, the PG2 fields are considered missing and may or may not be denoted as coming from the GLL grid. This does not appear to fully resolve until `initialize_atm_procs()` is called, but changes do appear to occur when `initialize_fields()` is called. I've got some example DAGs below to demonstrate this as it happens in the `homme_shoc_cld_spa_p3_rrtmgp_pg2_dp` test.

So, my thought is that regenerating and destroying, while non-optimal, is the simpler option because it's conducted  during initialization and should be relatively low in memory and cycles. I'm certainly willing to be persuaded otherwise, though--will just require a little retooling.

![scream_atm_createField_dag](https://github.com/user-attachments/assets/929e0e2c-4b31-4f7c-a609-5612d2c6fe73)
![scream_atm_initField_dag](https://github.com/user-attachments/assets/7e4069d4-5ea1-4b6e-b836-7e4f0b8003d0)
![scream_atm_initProc_dag](https://github.com/user-attachments/assets/1369d4c3-e2a2-4765-a835-34d3c112377e)